multiaddr_test.go

```diff
@@ -215,12 +215,6 @@ func TestBytesSplitAndJoin(t *testing.T) {
 			t.Errorf("joined components failed: %s != %s", m, joined)
 		}
 
-		// modifying underlying bytes is fine.
-		m2 := m.(*multiaddr)
-		for i := range m2.bytes {
```

Comment:
could change this to just directly modify m2. I think the check is still valuable.

Reply:
No, this is actually broken, that's why I removed it. I'm no longer copying on `Split`, just slicing. I'm now assuming that multiaddrs are immutable. I'm not entirely sure why we had this test in the first place.